File: src/main/java/org/demoth/aworlds/server2/api/messaging/fromClient/JoinMessage.java
package org.demoth.aworlds.server2.api.messaging.fromClient;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.demoth.aworlds.server2.api.messaging.Message;

public class JoinMessage extends Message {
    @JsonProperty("character_id")
    public String characterId;

    public JoinMessage() {
    }

    public JoinMessage(String characterId) {
        this.characterId = characterId;
    }
}

File: src/main/kotlin/org/demoth/aworlds/server2/api/messaging/fromClient/JoinMessage.kt
package org.demoth.aworlds.server2.api.messaging.fromClient

import com.fasterxml.jackson.annotation.JsonProperty
import org.demoth.aworlds.server2.api.messaging.Message

class JoinMessage(
        @JsonProperty("character_id") var characterId: String) : Message() {

    override fun toString(): String {
        return "JoinMessage{" +
                "characterId='" + characterId + '\''.toString() +
                '}'.toString()
    }
}
